app/logic/stake_logic.rb

```diff
@@ -198,10 +198,24 @@ def count_average_validators_commission
 
       StakePool.where(network: p.payload[:network]).each do |pool|
         validator_ids = pool.stake_accounts.active.pluck(:validator_id)
-        average_commission = ValidatorScoreV1.where(validator_id: validator_ids)
-                                             .average(:commission)
+        total_active_stake = 0
+        commissions = []
+
+        Validator.where(id: validator_ids)
+                 .includes(:stake_accounts, :validator_score_v1)
+                 .find_each do |validator|
```

Comment:
I'd look for a way to refactor this. There's a lot of shared logic in this method and in `update_validator_stats` (pulling exactly the same validators from db, calculating total_active_stake again etc.).
Otherwise, we'll add another big query to db here.
Perhaps extract this logic to a separate method called before `update_validator_stats` and `count_average_validators_commission`.